Added button to visualise key change

Segebre opened the pull request:
While working on the exercises, I wanted to see the key update in action as I found the whole concept very interesting. This is the way I did it and thought it would be nice to share it in the final version to help others out; this way, others can also visualise the key change and better solidify the explanation from the videos.

kentcdodds commented:
Thanks for this. Maybe you could instead add an example to the examples directory so we don't change this code from what it is in the videos?

Segebre commented:
@kentcdodds yes, that seems better. I have reverted the previous changes, moved the example to the `examples` folder as requested, and added a comment on top of the `useEffect` hook which handles the key change in localStorage. Please let me know if the file name and comments are suitable. And thank you for taking the time to review this PR!

kentcdodds commented on src/examples/local-state-key-change.js:
```diff
@@ -0,0 +1,74 @@
+
+import * as React from 'react'
+
+function useLocalStorageState(
```
Instead of hard-coding this, could we import it from [here](https://github.com/kentcdodds/react-hooks/blob/861594b4d999368c037604ad32e222f14d15465d/src/utils.js#L10)?

Segebre replied:
Yes! I completely missed this utils function as I have not yet gotten there. Great suggestion and thank you!

kentcdodds commented on src/examples/local-state-key-change.js:
```diff
@@ -0,0 +1,74 @@
+  )
+}
+
+function App() {
+  return <Greeting />
+}
+
+export default App
```
We can simplify this because it's just an example

```suggestion
export default Greeting
```

kentcdodds commented on src/examples/local-state-key-change.js:
```diff
@@ -0,0 +1,74 @@
+// useEffect: persistent state
+// 💯 flexible localStorage hook - changing the key in localStorage
```
We don't need 💯 because it's not an extra credit.

```suggestion
// flexible localStorage hook - changing the key in localStorage
```

kentcdodds reviewed (commented):
Thanks! Just a few things.

kentcdodds reviewed (approved):
Super! Thank you 👏

kentcdodds commented:
@all-contributors please add @Segebre for code
